assets/js/atomic/blocks/product-elements/image/block.tsx

```diff
@@ -17,39 +16,74 @@ import {
 } from '@woocommerce/base-hooks';
 import { withProductDataContext } from '@woocommerce/shared-hocs';
 import { useStoreEvents } from '@woocommerce/base-context/hooks';
+import type { HTMLAttributes } from 'react';
 
 /**
  * Internal dependencies
  */
 import ProductSaleBadge from '../sale-badge/block';
 import './style.scss';
+import type { BlockAttributes } from './types';
 
-/**
- * Product Image Block Component.
- *
- * @param {Object}            props                           Incoming props.
- * @param {string}            [props.className]               CSS Class name for the component.
- * @param {string|undefined}  [props.imageSizing]             Size of image to use.
- * @param {boolean|undefined} [props.showProductLink]         Whether or not to display a link to the product page.
- * @param {boolean}           [props.showSaleBadge]           Whether or not to display the on sale badge.
- * @param {string|undefined}  [props.saleBadgeAlign]          How should the sale badge be aligned if displayed.
- * @param {boolean}           [props.isDescendentOfQueryLoop] Whether or not be a children of Query Loop Block.
- * @return {*} The component.
- */
-export const Block = ( props ) => {
+const ImagePlaceholder = (): JSX.Element => {
+	return (
+		<img src={ PLACEHOLDER_IMG_SRC } alt="" width={ 500 } height={ 500 } />
+	);
+};
+
+type ImageProps = {
+	image: {
+		alt?: string;
+		id: number;
+		name: string;
+		sizes: string;
+		src: string;
+		srcset: string;
+		thumbnail: string;
+	};
+	loaded: boolean;
+	showFullSize: boolean;
+	fallbackAlt: string;
+};
+
+const Image = ( {
+	image,
+	loaded,
+	showFullSize,
+	fallbackAlt,
+}: ImageProps ): JSX.Element => {
+	const { thumbnail, src, srcset, sizes, alt } = image || {};
+	const imageProps = {
+		alt: alt || fallbackAlt,
+		hidden: ! loaded,
+		src: thumbnail,
+		...( showFullSize && { src, srcSet: srcset, sizes } ),
```

Comment:
`src`, `srcSet` and `sizes` only seem to be required when `showFullSize` is `true`, right? Or we want to make sure that these are always provided to the component?

Reply:
As these props optional, I marked them accordingly. Thanks for pointing this out, @sunyatasattva.